### ISSUE TITLE
mb_parse_str() attempts to return a void string from parse_str()

### ISSUE THREAD

**Issue by Donatello-za** [tchwork/utf8#36]
The mb_parse_str() function in Utf8/Bootup/mbstring.php returns with a result from parse_str() but parse_str() itself actually returns a void. This produces code inspection warnings and possible bugs.

**Comment by nicolas-grekas** [comment]
Which tool did you use to get this warning? I fail to see the relevancy of this rule...

**Comment by Donatello-za** [comment]
PHPStorm picks up the issue I mentioned and the reason why it is a potential problem is that a developer might think that your mb_parse_str() function actually returns the array of variables instead of it begin passed by reference. Your mb_parse_str function should not return with anything, i.e:

Instead of this:
```php
function mb_parse_str($s, &$result = array()) {return parse_str($s, $result);};
```
It should be:
```php
function mb_parse_str($s, &$result = array()) {parse_str($s, $result);};
````

**Comment by nicolas-grekas** [comment]
Closed via 55ec83c
Thank you @Donatello-za
